adoc/chapters/programming_interface.adoc

```diff
@@ -13786,8 +13677,8 @@ void copy(accessor<SrcT, SrcDims,
 ----
 a@ Copies the contents of the memory object accessed by
 [code]#src# into the memory pointed to by [code]#dest#.
-[code]#dest# must have at least as many bytes as the
-range accessed by [code]#src#.
+[code]#dest# must be a host pointer and must have at least
+as many bytes as the range accessed by [code]#src#.
```

Comment:
This is OK and it matches the intro paragraph to this section, but this is very restrictive and there are likely to be at least some cases where dest can be a pointer to a USM allocation.  It might be better to leave this section ambiguous until we can work out the details, or adopting text similar to memcpy.

(Same goes for the other copy cases below.)

Reply:
All the "copy" APIs that take an accessor were originally intended to copy between the host and a device.  I do not think the committee intended to expand these functions when it added USM.  Even if we wanted to expand them, I think it should not be done as part of this PR.  Therefore, I do not want this PR to expand the definition to say that `src` can be a USM pointer that is accessible on the handler's device.

My point in the meeting yesterday was that `src` should be allowed to be any pointer that is accessible on the host.  This would include, for example, any USM host or shared pointer.  However, since we agree that we are not ready for this, I left the wording vague by saying it must be a "host pointer".

Reply:
I'm OK with this for now, and it is easier to relax restrictions than it is to add them, but I do suspect we have code in the wild that is copying from an accessor to a USM allocation and therefore violating the updated description.

Reply:
Do you think the code "in the wild" is trying to copy between an accessor and a Device USM pointer (i.e. some USM memory that is **not** accessible on the host)?  This is the case that I think the committee did not intend to allow.

All other cases would be allowed by the wording I have in this PR if we interpret "host pointer" to mean "a pointer that is accessible on the host".

Reply:
I suspect the most common case by far will be copying to a host USM allocation.

Reply:
No change here ... I think we agreed that the current wording is OK for now.